code/iaas/allocator/src/main/java/io/cattle/platform/allocator/constraint/HostAffinityConstraint.java

```diff
@@ -33,16 +34,26 @@ public boolean matches(AllocationAttempt attempt,
         Set<Long> hostIds = candidate.getHosts();
         if (op == AffinityOps.SOFT_EQ || op == AffinityOps.EQ) {
             for (Long hostId : hostIds) {
-                Map<String, String> labelsForHost = allocatorDao.getLabelsForHost(hostId);
-                if (!labelValue.equals(labelsForHost.get(labelKey))) {
+                Map<String, String[]> labelsForHost = allocatorDao.getLabelsForHost(hostId);
```

Comment:
The map is now key: [value, hostLabelMap.state].  It's a little bit strange of a data structure, but we needed the additional state info for the hostLabelMap.